fenicsprecice/fenicsprecice.py

```diff
@@ -44,19 +48,32 @@ def __init__(self, adapter_config_filename='precice-adapter-config.json'):
 
         self._config = Config(adapter_config_filename)
 
-        self._interface = precice.Interface(self._config.get_participant_name(), self._config.get_config_file_name(), 0,
-                                            1)
+        # Setup up MPI communicator on mpi4py
+        self._comm = MPI.COMM_WORLD
+        self._rank = self._comm.Get_rank()
+        self._size = self._comm.Get_size()
+
+        self._interface = precice.Interface(self._config.get_participant_name(), self._config.get_config_file_name(),
+                                            self._rank, self._size)
 
         # FEniCS related quantities
-        self._fenics_dimensions = None
         self._read_function_space = None  # initialized later
+        self._write_function_space = None  # initialized later
+        self._dofmap = None  # initialized later using function space provided by user
+        self._fenics_gids = None
+        self._fenics_coords = None
+        self._coupling_subdomain = None
 
         # coupling mesh related quantities
-        self._coupling_mesh_vertices = None  # initialized later
+        self._owned_lids = None
+        self._owned_gids = None  # initialized later
+        self._owned_coords = None  # initialized later
```

Comment:
```suggestion
        self._owned_coupling_vertices_local_ids = None
        self._owned_coupling_vertices_global_ids = None  # initialized later
        self._owned_coupling_vertices_coordinates = None  # initialized later
```

Possible refactoring: Collect this under `self._owned_coupling_vertices`.